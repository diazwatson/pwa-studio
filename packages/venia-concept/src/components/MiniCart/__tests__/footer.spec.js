import React from 'react';
import ShallowRenderer from 'react-test-renderer/Shallow';

import Footer from '../footer';

const renderer = new ShallowRenderer();

const baseProps = {
    cart: {
        isEditingItem: false,
        isLoading: false
    },
    isCartEmpty: false,
    isMiniCartMaskOpen: false
};

test('renders totals summary and checkout components', () => {
    const tree = renderer.render(<Footer {...baseProps} />);

    expect(tree).toMatchSnapshot();
});

test('returns null when cart is empty', () => {
    const props = {
        ...baseProps,
        isCartEmpty: true
    };
    const tree = renderer.render(<Footer {...props} />);

    expect(tree).toMatchSnapshot();
});

test('returns null when cart is loading', () => {
    const props = {
        ...baseProps,
        cart: {
            ...baseProps.cart,
            isLoading: true
        }
    };
    const tree = renderer.render(<Footer {...props} />);

    expect(tree).toMatchSnapshot();
});

test('returns null when cart is editing item', () => {
    const props = {
        ...baseProps,
        cart: {
            ...baseProps.cart,
            isEditingItem: true
        }
    };
    const tree = renderer.render(<Footer {...props} />);

    expect(tree).toMatchSnapshot();
});
