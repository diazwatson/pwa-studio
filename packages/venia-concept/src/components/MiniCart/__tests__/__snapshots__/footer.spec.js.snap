// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders totals summary and checkout components 1`] = `
<div>
  <TotalsSummary
    cart={
      Object {
        "isEditingItem": false,
        "isLoading": false,
      }
    }
  />
  <React.Suspense
    fallback={
      <div>
        <CheckoutButton
          ready={false}
        />
      </div>
    }
  >
    <Connect(CheckoutWrapper)
      cart={
        Object {
          "isEditingItem": false,
          "isLoading": false,
        }
      }
    />
  </React.Suspense>
</div>
`;

exports[`returns null when cart is editing item 1`] = `null`;

exports[`returns null when cart is empty 1`] = `null`;

exports[`returns null when cart is loading 1`] = `null`;
